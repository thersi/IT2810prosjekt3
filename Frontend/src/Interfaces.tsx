import {
  Dispatch,
  SetStateAction,
} from "react-transition-group/node_modules/@types/react";

export interface Movie {
  _id: string;
  title: string;
  thumbsUp: number;
  year: number;
  genre: string[];
  actors: string[];
  thumbsDown: number;
  poster: string;
}

export interface MovieAppBarProps {
  handleSearch: (value: string) => void;
  handleFilter: (value: string) => void;
  handleSort: (value: boolean) => void;
}

export interface ThumbsByIdInput {
  thumbsUpByIdId: string;
}

export interface ThumbsDownByIdInput {
  thumbsDownByIdId: string;
}

export interface ThumbsUpByIdResult {
  thumbsUpById: Movie;
}

export interface ThumbsDownByIdResult {
  thumbsDownById: Movie;
}

export interface MovieDialogProps {
  movie: Movie;
  setOpen: Dispatch<SetStateAction<boolean>>;
  setThumbsUp: Dispatch<SetStateAction<number>>;
  thumbsUp: number;
  setThumbsDown: Dispatch<SetStateAction<number>>;
  thumbsDown: number;
  voted: boolean;
  setVoted: Dispatch<SetStateAction<boolean>>;
}

export interface QueryMoviesResult {
  searchAndFilter: Movie[];
}

export interface QueryMoviesInput {
  filterGenre: string;
  limit: number;
  page: number;
  order: number;
  sortOn: string;
  word: string;
}

export interface MovieByIdInput {
  movieByIdId: string;
}

export interface MovieByIdResult {
  movieById: Movie;
}

export interface MovieListProps {
  data: QueryMoviesResult;
}
