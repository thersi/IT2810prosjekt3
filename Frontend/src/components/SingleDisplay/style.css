
.movie {
  display: flex;
  padding: 3px;
  border: 2px outset;
  border-color: rgba(46, 140, 240, 0.1);
  background-color: rgba(46, 140, 240, 0.1);
  color: rgb(19, 48, 78);
  width: 70%;
  font-family: Arial, Helvetica, sans-serif;
}
@media screen and (max-width: 600px) {
  .movie {
    width: 95%;
  }
}

.movieInfo {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 3px;
}

img {
  text-align: center;
  flex-basis: 40%;
  display: block;
  width: 120px;
  height: 160px;
}

.text {
  padding-left: 2%;
  font-size: 20px;
  flex-direction: column;
  display: block;
  width: 100%;
  height: 100%;
  padding-top: 3%;
}

.thumbButtons {
  display: flex;
  flex-direction: flex-end;

}
.thumbUp {
  padding: 20pt;
  color: rgb(125, 143, 125);
}
.thumbDown {
  padding: 20pt;
  color: rgb(175, 131, 131);
}
