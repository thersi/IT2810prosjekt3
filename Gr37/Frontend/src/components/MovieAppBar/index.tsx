import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import MenuIcon from "@mui/icons-material/Menu";
import SearchBar from "../SearchBar";
import MovieFilter from "../MovieFilter";
import { createTheme } from "@mui/system";

export default function MovieAppBar() {
  const darkTheme = createTheme({
    palette: {
      mode: "dark",
    },
  });
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
          >
            Movies
          </Typography>
          <MovieFilter />
          <SearchBar />
        </Toolbar>
      </AppBar>
    </Box>
  );
}
