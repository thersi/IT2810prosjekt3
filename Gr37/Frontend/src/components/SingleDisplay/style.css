.movie {
  display: flex;
  align-items: left;
  justify-content: left;
}

img {
  text-align: center;
  flex-basis: 40%;
  display: block;
  max-width: 90%;
}

.text {
  font-size: 20px;
  flex-direction: column;
  display: block;
}
