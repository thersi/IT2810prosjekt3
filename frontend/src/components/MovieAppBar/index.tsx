import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import MenuIcon from "@mui/icons-material/Menu";
import SearchBar from "../SearchBar";
import MovieFilter from "../MovieFilter";

export default function MovieAppBar() {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
          >
            Movies
          </Typography>
          <MovieFilter />
          <SearchBar />
        </Toolbar>
      </AppBar>
    </Box>
  );
}
