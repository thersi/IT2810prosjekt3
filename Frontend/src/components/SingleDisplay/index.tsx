import { useState } from "react";
import MovieDialog from "../MovieDialog";
import "./style.css";
import ThumbUpIcon from "@material-ui/icons/ThumbUp";
import ThumbDownIcon from "@material-ui/icons/ThumbDown";
import {
  MovieByIdInput,
  MovieByIdResult,
  SingleDisplayProps,
} from "../../Interfaces/Interfaces";
import { useLazyQuery } from "@apollo/client";
import { QUERY_MOVIE_BY_ID } from "../../Queries/queries";

export default function SingleDisplay(props: SingleDisplayProps) {
  const { movie, refetch } = props;
  const [open, setOpen] = useState(false);
  const [thumbsUp, setThumbsUp] = useState(movie.thumbsUp);
  const [thumbsDown, setThumbsDown] = useState(movie.thumbsDown);

  const vote = sessionStorage.getItem(movie._id);
  const dflt_vote = vote !== null ? JSON.parse(vote) : false;
  const [voted, setVoted] = useState(dflt_vote);

  const [fetchMovie, { data: movieByIdData, loading: movieByIdLoading }] =
    useLazyQuery<MovieByIdResult, MovieByIdInput>(QUERY_MOVIE_BY_ID);

  if (movieByIdLoading) {
    return <p>Movie is loading</p>;
  }

  if (open && typeof movieByIdData !== "undefined") {
    return (
      <MovieDialog
        movie={movieByIdData.movieById}
        setOpen={setOpen}
        setThumbsUp={setThumbsUp}
        thumbsUp={thumbsUp}
        setThumbsDown={setThumbsDown}
        thumbsDown={thumbsDown}
        voted={voted}
        setVoted={setVoted}
        refetch={refetch}
      />
    );
  }

  return (
    <div
      className="movie"
      onClick={() => {
        setOpen(true);
        fetchMovie({ variables: { movieByIdId: movie._id } });
      }}
    >
      <div>
        <img className="image" src={movie.poster} alt="movie"></img>
      </div>
      <div className="movieInfo">
        <div className="text">
          {<label>{movie.title + " (" + movie.year + ") "}</label>}
        </div>
        <div className="thumbButtons">
          {" "}
          <div className="thumbUp">
            <ThumbUpIcon />
            {thumbsUp}
          </div>
          <div className="thumbDown">
            <ThumbDownIcon />
            {thumbsDown}
          </div>
        </div>
      </div>
    </div>
  );
}
