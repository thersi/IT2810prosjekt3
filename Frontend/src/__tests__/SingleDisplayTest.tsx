import EnzymeToJson from "enzyme-to-json";
import { configure, mount, render, shallow } from "enzyme";
import MovieAppBar from "../components/MovieAppBar";
import Adapter from "enzyme-adapter-react-16";
import SingleDisplay from "../components/SingleDisplay";
import { Movie } from "../Interfaces/Interfaces";

const movie: Movie = {
  _id: "6169c636b56d4eff214f61f6",
  title: "Avengers: Endgame",
  year: 2019,
  thumbsUp: 196,
  thumbsDown: 37,
  genre: ["action", "comedy"],
  actors: ["Robert Downey Jr.", "Chris Evans"],
  poster: "url",
};

const component = <SingleDisplay key={0} movie={movie} />;
configure({ adapter: new Adapter() });
describe("Single Movie Display", () => {
  it("renders correctly", () => {
    render(component);
  });
  it("Should match snapshot", () => {
    const snapshotCheck = shallow(component);
    expect(EnzymeToJson(snapshotCheck)).toMatchSnapshot();
  });

  it("Should display correct movie information", () => {
    const shallowComp = mount(component);
    expect(shallowComp.find(".thumbUp").contains(196)).toBe(true);
    expect(shallowComp.find(".thumbDown").contains(37)).toBe(true);
    expect(shallowComp.find(".thumbDown").contains(89)).toBe(false);
    expect(shallowComp.find(".thumbDown").contains(89)).toBe(false);
    expect(
      shallowComp.find(".text").contains("Avengers: Endgame (2019) ")
    ).toBe(true);
  });
});
