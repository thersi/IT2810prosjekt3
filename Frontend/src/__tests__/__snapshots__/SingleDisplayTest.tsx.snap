// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Movie Display Should match snapshot 1`] = `
<div
  className="movie"
  onClick={[Function]}
>
  <div>
    <img
      alt="movie"
      className="image"
      src="url"
    />
  </div>
  <div
    className="movieInfo"
  >
    <div
      className="text"
    >
      <label>
        Avengers: Endgame (2019) 
      </label>
    </div>
    <div
      className="thumbButtons"
    >
       
      <div
        className="thumbUp"
      >
        <Memo(ForwardRef(ThumbUpIcon)) />
        196
      </div>
      <div
        className="thumbDown"
      >
        <Memo(ForwardRef(ThumbDownIcon)) />
        37
      </div>
    </div>
  </div>
</div>
`;
